types/underscore/underscore-tests.ts

```diff
@@ -1186,6 +1152,54 @@ declare const extractChainTypes: ChainTypeExtractor;
+
+    // function iteratee - dictionaries
+    _.reject(stringRecordDictionary, stringRecordDictionaryBooleanIterator, context); // $ExpectType StringRecord[]
+    _(stringRecordDictionary).reject(stringRecordDictionaryBooleanIterator, context); // $ExpectType StringRecord[]
+    extractChainTypes(_.chain(stringRecordDictionary).reject(stringRecordDictionaryBooleanIterator, context)); // $ExpectType ChainType<StringRecord[], StringRecord>
+
+    // function iteratee - strings
+    _.reject(simpleString, stringListBooleanIterator, context); // $ExpectType string[]
+    _(simpleString).reject(stringListBooleanIterator, context); // $ExpectType string[]
+    extractChainTypes(_.chain(simpleString).reject(stringListBooleanIterator, context)); // $ExpectType ChainType<string[], string>
+
+    // partial object iteratee - lists
+    _.reject(stringRecordList, partialStringRecord); // $ExpectType StringRecord[]
+    _(stringRecordList).reject(partialStringRecord); // $ExpectType StringRecord[]
+    extractChainTypes(_.chain(stringRecordList).reject(partialStringRecord)); // $ExpectType ChainType<StringRecord[], StringRecord>
+
+    // partial object iteratee - dictionaries
+    _.reject(stringRecordDictionary, partialStringRecord); // $ExpectType StringRecord[]
+    _(stringRecordDictionary).reject(partialStringRecord); // $ExpectType StringRecord[]
+    extractChainTypes(_.chain(stringRecordDictionary).reject(partialStringRecord)); // $ExpectType ChainType<StringRecord[], StringRecord>
+
+    // property name iteratee - lists
+    _.reject(stringRecordList, stringRecordProperty); // $ExpectType StringRecord[]
+    _(stringRecordList).reject(stringRecordProperty); // $ExpectType StringRecord[]
+    extractChainTypes(_.chain(stringRecordList).reject(stringRecordProperty)); // $ExpectType ChainType<StringRecord[], StringRecord>
+
+    // property name iteratee - dictionaries
+    _.reject(stringRecordDictionary, stringRecordProperty); // $ExpectType StringRecord[]
+    _(stringRecordDictionary).reject(stringRecordProperty); // $ExpectType StringRecord[]
+    extractChainTypes(_.chain(stringRecordDictionary).reject(stringRecordProperty)); // $ExpectType ChainType<StringRecord[], StringRecord>
+
+    // property path iteratee - lists
+    _.reject(stringRecordList, stringRecordPropertyPath); // $ExpectType StringRecord[]
+    _(stringRecordList).reject(stringRecordPropertyPath); // $ExpectType StringRecord[]
+    extractChainTypes(_.chain(stringRecordList).reject(stringRecordPropertyPath)); // $ExpectType ChainType<StringRecord[], StringRecord>
+
+    // property path iteratee - dictionaries
+    _.reject(stringRecordDictionary, stringRecordPropertyPath); // $ExpectType StringRecord[]
+    _(stringRecordDictionary).reject(stringRecordPropertyPath); // $ExpectType StringRecord[]
+    extractChainTypes(_.chain(stringRecordDictionary).reject(stringRecordPropertyPath)); // $ExpectType ChainType<StringRecord[], StringRecord>
+}
```

Comment:
I'm getting used to this way of testing. I'm even starting to like it.

Perhaps do add a few cases (just a sampling) with an implicit `_.identity` iteratee.

Reply:
I added the identity iteratee tests and also switched to alternating between testing lists and dictionaries for non-function iteratee tests for find, filter, and reject since I feel like testing both of those in those cases is not super interesting. Let me know if you disagree with that, though.

Reply:
I'm fine with it. I'm a bit surprised that you choose to do sampling after so much combinatorial completeness, though. ;-)

Reply:
Just trying to find ways to stick to the middle ground where we're both somewhat happy 😄